main.go

```diff
@@ -39,13 +40,22 @@ func main() {
 	flag.Parse()
 	log.InitLogger(*logFile, *logLevel)
 
+	// When enableAuthzPolicyController is set to true create a dry run folder which
+	// would contain the Authorization Policy resource for all the namespaces/services which
+	// are not passed as a parameter in --ap-enabled-list
 	if *enableAuthzPolicyController {
-		if _, err := os.Stat(common.DryRunStoredFilesDirectory); os.IsNotExist(err) {
-			err := os.MkdirAll(common.DryRunStoredFilesDirectory, 0755)
+		// If the Authz Policy folder already exists remove the stale data
+		if _, err := os.Stat(common.DryRunStoredFilesDirectory); !os.IsNotExist(err) {
```

Comment:
can you switch to os.IsExist()?

Reply:
`os.IsExist(err)` would not work here. `os.IsNotExist(err)` & `os.IsExist(err)` both rely on the err message from `os.Stat` to determine if the file exists or not. In case the file exists `os.Stat` will return nil and `os.IsExist` would thus return false (as err is nil). Thus, we make use of `!os.IsNotExist(err)`.

Reply:
Actually, let me change that to `err==nil`